modules/cas_disk/exp_obj.c

```diff
@@ -530,36 +530,29 @@ int casdsk_exp_obj_create(struct casdsk_disk *dsk, const char *dev_name,
 	exp_obj->owner = owner;
 	exp_obj->ops = ops;
 
-	gd = alloc_disk(1);
-	if (!gd) {
-		result = -ENOMEM;
+	result = _casdsk_init_tag_set(dsk, &dsk->tag_set);
+	if (result) {
+		goto error_init_tag_set;
+	}
+
+	result = cas_alloc_mq_disk(&gd, &queue, &dsk->tag_set);
+	if (result) {
 		goto error_alloc_disk;
 	}
+
 	exp_obj->gd = gd;
 
 	result = _casdsk_exp_obj_set_dev_t(dsk, gd);
 	if (result)
 		goto error_dev_t;
```

Comment:
Since `cas_alloc_mq_disk()` succeeded a few lines above, you need to call `blk_cleanup_queue()` in rollback